[Bug]: n8n error

### Runtime environment

4.5.0

### Exception

[2025-11-13 16:16:23 +0800] [10] [INFO] 10.1.9.27:52101 GET /api/v1/pipelines/_/metadata 1.1 200 23413 4169
[2025-11-13 16:16:23 +0800] [10] [INFO] 10.1.9.27:51278 GET /api/v1/pipelines/94bda9a8-5eee-4c6b-a08d-dbe8ce993006 1.1 200 2063 5555
[2025-11-13 16:16:27 +0800] [10] [INFO] 10.1.9.27:51278 GET /api/v1/pipelines/94bda9a8-5eee-4c6b-a08d-dbe8ce993006/chat/messages/person 1.1 200 1527 3628
[2025-11-13 16:16:27 +0800] [10] [INFO] 10.1.9.27:51278 GET /api/v1/pipelines/94bda9a8-5eee-4c6b-a08d-dbe8ce993006/chat/messages/person 1.1 200 1527 1431
[11-13 16:16:31.271] process.py (39) - [INFO] : Processing request from person_webchatperson (16): testsddd
[11-13 16:16:31.616] chat.py (114) - [ERROR] : 对话(16)请求失败: ValueError "Message" object has no field "resp_message_id"
Traceback (most recent call last):
  File "/app/pkg/pipeline/process/handlers/chat.py", line 82, in handle
    result.resp_message_id = str(resp_message_id)
    ^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/pydantic/main.py", line 1032, in __setattr__
    elif (setattr_handler := self._setattr_handler(name, value)) is not None:
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/pydantic/main.py", line 1079, in _setattr_handler
    raise ValueError(f'"{cls.__name__}" object has no field "{name}"')
ValueError: "Message" object has no field "resp_message_id"
[11-13 16:16:31.617] pipelinemgr.py (236) - [ERROR] : Error processing query 16 stage=MessageProcessor : list index out of range
[11-13 16:16:31.618] pipelinemgr.py (237) - [ERROR] : Traceback: Traceback (most recent call last):
  File "/app/pkg/pipeline/pipelinemgr.py", line 233, in process_query
    await self._execute_from_stage(0, query)
  File "/app/pkg/pipeline/pipelinemgr.py", line 193, in _execute_from_stage
    await self._check_output(query, sub_result)
  File "/app/pkg/pipeline/pipelinemgr.py", line 120, in _check_output
    bot_message=query.resp_messages[-1],
                ~~~~~~~~~~~~~~~~~~~^^^^
IndexError: list index out of range

### Reproduction steps

response body is extremely simple.
```
{
  "resp_message_id":"001",
  "response": "hello calvin"
}
```

### Enabled plugins

none

--- a/src/langbot/pkg/provider/runners/n8nsvapi.py
+++ b/src/langbot/pkg/provider/runners/n8nsvapi.py
@@ -68,6 +68,33 @@ class N8nServiceAPIRunner(runner.RequestRunner):
 
         return plain_text
 
+    async def _process_stream_response(self, response: aiohttp.ClientResponse) -> typing.AsyncGenerator[
+        provider_message.Message, None]:
+        """处理流式响应"""
+        full_content = ""
+        message_idx = 0
+        is_final = False
+        async for chunk in response.content.iter_chunked(1024):
+            if not chunk:
+                continue
+
+            try:
+                data = json.loads(chunk)
+                if data.get('type') == 'item' and 'content' in data:
+                    message_idx += 1
+                    content = data['content']
+                    full_content += content
+                elif data.get('type') == 'end':
+                    is_final = True
+                if is_final or message_idx % 8 == 0:
+                    yield provider_message.MessageChunk(
+                        role='assistant',
+                        content=full_content,
+                        is_final=is_final,
+                    )
+            except json.JSONDecodeError:
+                self.ap.logger.warning(f"Failed to parse final JSON line: {response.text()}")
+
     async def _call_webhook(self, query: pipeline_query.Query) -> typing.AsyncGenerator[provider_message.Message, None]:
         """调用n8n webhook"""
         # 生成会话ID（如果不存在）
@@ -80,6 +107,7 @@ class N8nServiceAPIRunner(runner.RequestRunner):
         # 准备请求数据
         payload = {
             # 基本消息内容
+            'chatInput' :plain_text,  # 考虑到之前用户直接用的message model这里添加新键
             'message': plain_text,
             'user_message_text': plain_text,
             'conversation_id': query.session.using_conversation.uuid,
@@ -131,37 +159,57 @@ class N8nServiceAPIRunner(runner.RequestRunner):
 
             # 调用webhook
             async with aiohttp.ClientSession() as session:
-                async with session.post(
-                    self.webhook_url, json=payload, headers=headers, auth=auth, timeout=self.timeout
-                ) as response:
-                    if response.status != 200:
-                        error_text = await response.text()
-                        self.ap.logger.error(f'n8n webhook call failed: {response.status}, {error_text}')
-                        raise Exception(f'n8n webhook call failed: {response.status}, {error_text}')
-
-                    # 解析响应
-                    response_data = await response.json()
-                    self.ap.logger.debug(f'n8n webhook response: {response_data}')
-
-                    # 从响应中提取输出
-                    if self.output_key in response_data:
-                        output_content = response_data[self.output_key]
-                    else:
-                        # 如果没有指定的输出键，则使用整个响应
-                        output_content = json.dumps(response_data, ensure_ascii=False)
-
-                    # 返回消息
                     if is_stream:
-                        yield provider_message.MessageChunk(
-                            role='assistant',
-                            content=output_content,
-                            is_final=True,
-                        )
+                        # 流式请求
+                        async with session.post(
+                                self.webhook_url,
+                                json=payload,
+                                headers=headers,
+                                auth=auth,
+                                timeout=self.timeout
+                        ) as response:
+                            if response.status != 200:
+                                error_text = await response.text()
+                                self.ap.logger.error(f'n8n webhook call failed: {response.status}, {error_text}')
+                                raise Exception(f'n8n webhook call failed: {response.status}, {error_text}')
+
+                            # 处理流式响应
+                            async for chunk in self._process_stream_response(response):
+                                yield chunk
                     else:
-                        yield provider_message.Message(
-                            role='assistant',
-                            content=output_content,
-                        )
+                        async with session.post(
+                                self.webhook_url,
+                                json=payload,
+                                headers=headers,
+                                auth=auth,
+                                timeout=self.timeout
+                        ) as response:
+                            try:
+                                async for chunk in self._process_stream_response(response):
+                                    output_content = chunk.content if chunk.is_final else ''
+                            except:
+                                # 非流式请求（保持原有逻辑）
+                                if response.status != 200:
+                                    error_text = await response.text()
+                                    self.ap.logger.error(f'n8n webhook call failed: {response.status}, {error_text}')
+                                    raise Exception(f'n8n webhook call failed: {response.status}, {error_text}')
+
+                                # 解析响应
+                                response_data = await response.json()
+                                self.ap.logger.debug(f'n8n webhook response: {response_data}')
+
+                                # 从响应中提取输出
+                                if self.output_key in response_data:
+                                    output_content = response_data[self.output_key]
+                                else:
+                                    # 如果没有指定的输出键，则使用整个响应
+                                    output_content = json.dumps(response_data, ensure_ascii=False)
+
+                            # 返回消息
+                            yield provider_message.Message(
+                                role='assistant',
+                                content=output_content,
+                            )
         except Exception as e:
             self.ap.logger.error(f'n8n webhook call exception: {str(e)}')
             raise N8nAPIError(f'n8n webhook call exception: {str(e)}')

